### PR TITLE
py-jupyter-core add new version and setuptools dependency

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-jupyter-core/package.py [hunk 1]
```diff
@@ -10,8 +10,9 @@ class PyJupyterCore(PythonPackage):
     """Core Jupyter functionality"""
 
     homepage = "http://jupyter-core.readthedocs.io/"
-    url      = "https://github.com/jupyter/jupyter_core/archive/4.2.0.tar.gz"
+    url      = "https://pypi.io/packages/source/j/jupyter-core/jupyter_core-1.17.2.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyter-core/package.py]: I'm all for switching the download URL to PyPI, but you'll need to update all previous checksums and make sure that the version in the URL is present in the package:
```suggestion
    url      = "https://pypi.io/packages/source/j/jupyter-core/jupyter_core-4.6.0.tar.gz"
```

File: var/spack/repos/builtin/packages/py-jupyter-core/package.py [hunk 2]
```diff
@@ -25,4 +26,5 @@ class PyJupyterCore(PythonPackage):
     version('4.0',   sha256='2b491a5da687ffab4e5702e37d3cb8b6b184cd46520cae9c459bdb1fd144f026')
 
     depends_on('python@2.7:2.8,3.3:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyter-core/package.py]:
```suggestion
    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
```

**Comment by adamjstewart** [comment]
Ping @healther

**Comment by healther** [comment]
I opted to update the checksums

**Comment by healther** [comment]
Travis fail seems unrelated?

**Comment by adamjstewart** [comment]
Still two more pending requests

**Comment by healther** [comment]
done

File: var/spack/repos/builtin/packages/py-jupyter-core/package.py [hunk 3]
```diff
@@ -10,19 +10,21 @@ class PyJupyterCore(PythonPackage):
...
+    version('4.0.5', sha256='9f6581b827f56cfa1771d7b1bd8ecc1274afa7f6e3e1046b7e0d4e05d52bf6e8')
+    version('4.0.4', sha256='fcf45478025f34174943993947f51a41ad871ac998a14bf1cb87d8eb61e75c6d')
+    version('4.0.3', sha256='12258d8c593c53bb08e09f3da63a418d7cb5b5852b3d0ffa29639402f56dcbdb')
+    version('4.0.2', sha256='13a46b3c493ac63bd75048d6d2142cfc44258bc6c260d96c506f0214fcd78a70')
+    version('4.0.1', sha256='7c165f7de7a063596f8be1bcfc86e9ba6897e38baf24e8510514690963600122')
+    version('4.0.0', sha256='9025208cdfc40718c7e3ab62b5e17aacf68e3fc66e34ff21fe032d553620122a')
 
-    depends_on('python@2.7:2.8,3.3:')
+    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyter-core/package.py]:
```suggestion
    depends_on('py-setuptools', when='@4.5.0:', type='build')
```

**Comment by healther** [comment]
ping @adamjstewart

**Pull request merged**